src/java/org/apache/cassandra/io/util/FileUtils.java

```diff
@@ -67,23 +71,84 @@
     public static final boolean isCleanerAvailable;
     private static final AtomicReference<Optional<FSErrorHandler>> fsErrorHandler = new AtomicReference<>(Optional.empty());
 
+    private static Class clsDirectBuffer;
+    private static MethodHandle mhDirectBufferCleaner;
+    private static MethodHandle mhCleanerClean;
+    private static MethodHandle mhDirectBufferAddress;
+
     static
     {
         boolean canClean = false;
         try
         {
+            clsDirectBuffer = Class.forName("sun.nio.ch.DirectBuffer");
+            Method mDirectBufferCleaner = clsDirectBuffer.getMethod("cleaner");
+            mhDirectBufferCleaner = MethodHandles.lookup().unreflect(mDirectBufferCleaner);
+            Method mCleanerClean = mDirectBufferCleaner.getReturnType().getMethod("clean");
+            mhCleanerClean = MethodHandles.lookup().unreflect(mCleanerClean);
+            Method mDirectBufferAddress = clsDirectBuffer.getMethod("address");
+            mhDirectBufferAddress = MethodHandles.lookup().unreflect(mDirectBufferAddress);
+
             ByteBuffer buf = ByteBuffer.allocateDirect(1);
-            ((DirectBuffer) buf).cleaner().clean();
+            cleanerClean(buf);
             canClean = true;
         }
         catch (Throwable t)
         {
+            logger.info("Cannot initialize un-mmaper. Compacted data files will not be removed promptly.", t);
```

Comment:
The original log message here was rather awful. I'm not sure sure what actions an operator could take after seeing this, and `Compacted data files will not be removed promptly` is sufficiently vague.

Maybe we could say something like: `Cannot initialize optimized memory deallocator. Some data, both in-memory and on-disk, may live longer due to garbage collection."

Reply:
Sounds good